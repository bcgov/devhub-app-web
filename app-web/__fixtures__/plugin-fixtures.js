/*
Copyright 2018 Province of British Columbia

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at 

   http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.

Created by Patrick Simonian
*/
export const SOURCE_DEVHUB_GITHUB_QL_NODE = {
  id: 'd90c1b12-9055-580b-adb3-b29354871657',
  source: 'design-system',
  sourceName: 'Design System',
  sourcePath: 'https://github.com/bcgov/design-system/',
  resourcePath: '/design-system/README_123o8123',
  path: 'components/footer/README.md',
  owner: 'bcgov',
  labels: ['Design System', 'Components'],
  fileName: 'about.md',
  fileType: 'Markdown',
  pagePath: '/design-system/about_G0Z6oO3_3',
  originalSource:
    'https://github.com/bcgov/design-system/blob/master/components/footer/something/README.md',
  internal: {
    mediaType: 'text/markdown',
    type: 'SourceDevhubGithub',
    owner: 'gatsby-source-github-all',
    contentDigest: '1239087asdfjkas',
    content: '---\nsomething:true\n---\n# Hello world\n',
  },
  childMarkdownRemark: {
    id: 'c03ba7af-a099-5dbe-8aec-5922922e0fcc >>> MarkdownRemark',
    html: '<h1>Hello World</h1>',
    frontmatter: {
      something: 'true',
      _PARENT: 'c03ba7af-a099-5dbe-8aec-5922922e0fcc',
      parent: 'c03ba7af-a099-5dbe-8aec-5922922e0fcc',
      status: null,
    },
    headings: [
      {
        value: 'Hello World',
        depth: 1,
      },
    ],
    internal: {
      content:
        '---\ndescription: What the Design System is and how it works.\ntitle: About\n---\n\n## About\n\nThe Design System helps developers and designers build better digital products and services.\n\nIt’s a collection of digital resources and tools – including a library of reusable UI/interface components and service patterns. The system makes it easier and faster to build custom B.C. government websites and applications. \n\nComponents are collectively built by the government community, meet accessibility standards and are open for input and improvement.\n\nUse the Design System if you’re building custom B.C. government and applications. For more information about applying for external URLs, contact the B.C. government Joint Working Group.\n\n## The Team\n\nThe Design System is managed by a team of designers and developers and the government community. The system is a part of the DevHub, a project by the Developers’ Exchange that brings together UI/interface and back-end resources for digital teams.\n\nKelsey Singbeil – Team Lead, Service Design\n\nBryan Smith – Service Design Specialist\n\nOri Nevares – Service Design Specialist\n\n## The Community\n\nThe system is collectively built and maintained by the government community. It’s open source, which means that anybody can provide input, offer suggestions and get involved.\n\nShare your feedback to make this resource better for everyone.\n\n## How the system works\n\nHow might we improve people’s interaction with B.C. government digital products and services?\n\nThe system is a part of the answer.\n\nIt provides developers and designers with the tools they need to build great digital government products and services. The system is collectively built and managed, so all the tools and resources are the product of the community.\n\nAs part of this, we are committed to making all the tools and resources we create:\n\n**Maintained**  \nThe community and Design System management team will support the maintenance and development of new resources and tools. This means you can flag issues and they’ll be addressed, discuss rationale and directly chat with the management team whenever you need.\n\n**Collaborative**\nThe system is only valuable if the community finds it valuable. If a tool or resource isn’t meeting your needs, tell us why and work with us to make it better. We’ll also come to you – in your office, at your desk and share what’s new, ask what you need and share how you can get involved.\n\n**Open**\nThe system is built in GitHub and meant to be shared and improved by everyone. The code is open source and our work is in the open. We are transparent about how choices are made and tools are improved. Ask us…anything.\n\n**Contribute to the Design System**\n\n***Help with design***\n-   Suggest a component\n-   Improve documentation\n\n***Help with code***\n-  Code contribution guide\n-   Report an issue\n\n***Help with usability***\n-   Component discussions\n-   Provide research findings\n\n',
      type: 'MarkdownRemark',
      contentDigest: 'b186249c65e12e08344e31e289d99ab8',
      owner: 'gatsby-transformer-remark',
    },
  },
};

export const FILE_QL_NODE = {
  id: '/foo/Index.test.js absPath of file',
  internal: {
    contentDigest: '2a9e4d519252bf9f9ccc9a931d2d731e',
    mediaType: 'application/javascript',
    type: 'File',
    description: 'File "src/pages/Index.test.js"',
    owner: 'gatsby-source-filesystem',
  },
  birthtime: '2018-10-24T21:16:43.521Z',
  publicURL: '/static/Index.test-2a9e4d519252bf9f9ccc9a931d2d731e.js',
  ctime: '2018-10-24T21:16:43.522Z',
  mtime: '2018-10-24T21:16:43.522Z',
  atime: '2018-10-25T20:32:50.743Z',
  birthtimeMs: 1540415803521.444,
  ctimeMs: 1540415803521.575,
  mtimeMs: 1540415803521.575,
  atimeMs: 1540499570743.2075,
  sourceInstanceName: 'pages',
  absolutePath: '/foo/Index.test.js',
  relativePath: 'Index.test.js',
  relativeDirectory: '',
  extension: 'js',
  size: 1674,
  prettySize: '1.67 kB',
  name: 'Index.test',
  ext: '.js',
  base: 'Index.test.js',
  dev: 16777220,
  mode: 33188,
  nlink: 1,
  uid: 323756479,
  gid: 1839645156,
  blocks: 8,
  ino: 11739371,
  blksize: 4194304,
  rdev: 0,
};
