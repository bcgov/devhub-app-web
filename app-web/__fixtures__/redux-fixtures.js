/*
Copyright 2018 Province of British Columbia

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at 

   http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.

Created by Patrick Simonian
*/
import * as actions from '../src/store/actions/actionTypes';
export const ACTIONS = {
  LOAD_SIPHON_NODES: {
    type: actions.LOAD_SIPHON_NODES,
    payload: {
      nodes: [{ foo: 'bar' }, { foo: 'baz' }, { foo: 'foo' }],
    },
  },
  FILTER_SIPHON_NODES: {
    type: actions.FILTER_SIPHON_NODES,
    payload: {
      filteredBy: 'foo',
      value: 'bar',
    },
  },
  FILTER_SIPHON_NODES_BY_ALL: {
    type: actions.FILTER_SIPHON_NODES,
    payload: {
      filteredBy: 'foo',
      value: 'All',
    },
  },
  SELECT_FILTER_OPTION: {
    type: actions.SET_SELECTED_FILTER_OPTION,
    payload: {
      foo: 'bar',
    },
  },
  TOGGLE_MAIN_NAVIGATION_ON: {
    type: actions.TOGGLE_MAIN_NAVIGATION,
    payload: {
      toggled: true,
    },
  },
  TOGGLE_MAIN_NAVIGATION_OFF: {
    type: actions.TOGGLE_MAIN_NAVIGATION,
    payload: {
      toggled: false,
    },
  },
};

export const INITIAL_STATES = {
  SIPHON: {
    nodes: [],
    filteredNodes: [],
    groupBy: null,
    loading: false,
    error: false,
    messages: [],
  },
  UI: {
    selectedFilterOption: null,
    mainNavigationToggled: false,
  },
};
