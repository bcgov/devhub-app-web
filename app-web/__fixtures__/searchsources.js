/*
Copyright 2019 Province of British Columbia

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at 

   http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.

Created by Patrick Simonian
*/
// run time apollo query fixtures

export const ROCKET_CHAT = [
  {
    id: '1', 
    roomId: '2', author: 'apple', time: '2019-02-12', message: 'apples are clearly superior',
  },
  {
    id: '2', 
    roomId: '2', author: 'orange', time: '2019-02-12', message: 'Nice joke apple, are trying to perform some time of standup?',
  },
  {
    id: '3', 
    roomId: '2', author: 'apple', time: '2019-02-12', message: 'Don\'t play coy with me orange, you know I\'m telling the truth',
  },
  {
    id: '4', 
    roomId: '2', author: 'orange', time: '2019-02-12', message: 'I ain\'t never hurd of people curing scurvy with apples yo. Hook line and sinker dawg.',
  },
]