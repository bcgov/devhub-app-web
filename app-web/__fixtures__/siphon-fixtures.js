const IDS = {
  DESIGN_SYSTEM: 'collection-1',
  DEVHUB: 'collection-2',
};

// SIPHON NODES
export const DESIGN_SYSTEM_NODE_1 = {
  id: '1',
  parent: {
    id: IDS.DESIGN_SYSTEM,
  },
  _metadata: {
    position: '000.000.000',
  },
  attributes: {
    personas: ['Designer'],
  },
  source: {
    displayName: 'Design System',
  },
  resource: {
    path: '/design-system/Z1bGIqu',
    type: 'Components',
  },
  unfurl: {
    type: 'markdown',
    label1: '',
    data1: '',
    label2: '',
    data2: '',
    image: 'https://github.com/bcgov/design-system/blob/master/components/about/about.md?raw=true',
    title: 'About',
    description: 'What the Design System is and how it works.',
    author: 'ksingbeil',
  },
  childMarkdownRemark: {
    frontmatter: {
      pageOnly: false,
    },
  },
};

export const DESIGN_SYSTEM_NODE_2 = {
  id: '2',
  parent: {
    id: IDS.DESIGN_SYSTEM,
  },
  _metadata: {
    position: '000.000.001',
  },
  attributes: {
    personas: ['Designer'],
  },
  source: {
    displayName: 'Design System',
  },
  resource: {
    path: '/design-system/Z1bGIqu',
    type: 'Components',
  },
  unfurl: {
    type: 'markdown',
    label1: '',
    data1: '',
    label2: '',
    data2: '',
    image: 'https://github.com/bcgov/design-system/blob/master/components/about/about.md?raw=true',
    title: 'About',
    description: 'What the Design System is and how it works.',
    author: 'ksingbeil',
  },
  childMarkdownRemark: {
    frontmatter: {
      pageOnly: false,
    },
  },
};

export const DESIGN_SYSTEM_NODE_3 = {
  id: '3',
  parent: {
    id: IDS.DESIGN_SYSTEM,
  },
  _metadata: {
    position: '000.000.002',
  },
  attributes: {
    personas: ['Designer'],
  },
  source: {
    displayName: 'Design System',
  },
  resource: {
    path: '/design-system/Z1bGIqu',
    type: 'Components',
  },
  unfurl: {
    type: 'markdown',
    label1: '',
    data1: '',
    label2: '',
    data2: '',
    image: 'https://github.com/bcgov/design-system/blob/master/components/about/about.md?raw=true',
    title: 'About',
    description: 'What the Design System is and how it works.',
    author: 'ksingbeil',
  },
  childMarkdownRemark: {
    frontmatter: {
      pageOnly: false,
    },
  },
};

export const DESIGN_SYSTEM_NODE_4 = {
  id: '4',
  parent: {
    id: IDS.DESIGN_SYSTEM,
  },
  _metadata: {
    position: '000.000.003',
  },
  attributes: {
    personas: ['Developer'],
  },
  source: {
    displayName: 'Design System',
  },
  resource: {
    path: '/design-system/Z1bGIqu',
    type: 'Components',
  },
  unfurl: {
    type: 'markdown',
    label1: '',
    data1: '',
    label2: '',
    data2: '',
    image: 'https://github.com/bcgov/design-system/blob/master/components/about/about.md?raw=true',
    title: 'About',
    description: 'What the Design System is and how it works.',
    author: 'ksingbeil',
  },
  childMarkdownRemark: {
    frontmatter: {
      pageOnly: false,
    },
  },
};

export const DEVHUB_NODE_1 = {
  id: '5',
  parent: {
    id: IDS.DEVHUB,
  },
  _metadata: {
    position: '001.000.000',
  },
  attributes: {
    personas: ['Developer'],
  },
  source: {
    displayName: 'Design System',
  },
  resource: {
    path: '/design-system/Z1bGIqu',
    type: 'Documentation',
  },
  unfurl: {
    type: 'markdown',
    label1: '',
    data1: '',
    label2: '',
    data2: '',
    image: 'https://github.com/bcgov/design-system/blob/master/components/about/about.md?raw=true',
    title: 'About',
    description: 'What the Design System is and how it works.',
    author: 'ksingbeil',
  },
  childMarkdownRemark: {
    frontmatter: {
      pageOnly: false,
    },
  },
};

export const DEVHUB_NODE_2 = {
  id: '6',
  parent: {
    id: IDS.DEVHUB,
  },
  _metadata: {
    position: '001.000.001',
  },
  attributes: {
    personas: ['Developer'],
  },
  source: {
    displayName: 'Design System',
  },
  resource: {
    path: '/design-system/Z1bGIqu',
    type: 'Documentation',
  },
  unfurl: {
    type: 'markdown',
    label1: '',
    data1: '',
    label2: '',
    data2: '',
    image: 'https://github.com/bcgov/design-system/blob/master/components/about/about.md?raw=true',
    title: 'About',
    description: 'What the Design System is and how it works.',
    author: 'ksingbeil',
  },
  childMarkdownRemark: {
    frontmatter: {
      pageOnly: false,
    },
  },
};

export const DEVHUB_NODE_3 = {
  id: '7',
  parent: {
    id: IDS.DEVHUB,
  },
  _metadata: {
    position: '001.000.001',
  },
  attributes: {
    personas: ['Product Owner'],
  },
  source: {
    displayName: 'Design System',
  },
  resource: {
    path: '/design-system/Z1bGIqu',
    type: 'Documentation',
  },
  unfurl: {
    type: 'markdown',
    label1: '',
    data1: '',
    label2: '',
    data2: '',
    image: 'https://github.com/bcgov/design-system/blob/master/components/about/about.md?raw=true',
    title: 'About',
    description: 'What the Design System is and how it works.',
    author: 'ksingbeil',
  },
  childMarkdownRemark: {
    frontmatter: {
      pageOnly: false,
    },
  },
};

export const DESIGN_SYSTEM_NODES = [
  DESIGN_SYSTEM_NODE_3,
  DESIGN_SYSTEM_NODE_2,
  DESIGN_SYSTEM_NODE_1,
  DESIGN_SYSTEM_NODE_4,
];

export const DEVHUB_NODES = [DEVHUB_NODE_1, DEVHUB_NODE_2, DEVHUB_NODE_3];

// COLLECTIONS
export const DESIGN_SYSTEM_COLLECTION = {
  id: IDS.DESIGN_SYSTEM,
  type: 'default',
  title: 'Design System',
  description: 'baz',
  nodes: DESIGN_SYSTEM_NODES,
};

export const DESIGN_SYSTEM_COLLECTION_SORTED = {
  id: IDS.DESIGN_SYSTEM,
  type: 'default',
  title: 'Design System',
  description: 'baz',
  nodes: DESIGN_SYSTEM_NODES.sort((a, b) =>
    a._metadata.position.localeCompare(b._metadata.position),
  ),
};

export const DEVHUB_COLLECTION = {
  id: IDS.DEVHUB,
  type: 'default',
  title: 'Devhub',
  description: 'baz',
  nodes: DEVHUB_NODES,
};

export const SIPHON_NODES = DESIGN_SYSTEM_NODES.concat(DEVHUB_NODES);

export const COLLECTIONS = [DESIGN_SYSTEM_COLLECTION, DEVHUB_COLLECTION];
export const SORTED_COLLECTIONS = [DESIGN_SYSTEM_COLLECTION_SORTED, DEVHUB_COLLECTION];
// creating a set of filtered collections where there aren't any developers
export const FILTERED_COLLECTIONS = SORTED_COLLECTIONS.map(collection => ({
  ...collection,
  nodes: collection.nodes.filter(node => node.attributes.personas[0] === 'Developer'),
}));
