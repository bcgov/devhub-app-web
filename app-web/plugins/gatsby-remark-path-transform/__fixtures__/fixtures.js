const PARAGRAPH_AST = {
    'type': 'paragraph',
    'children': [
      {'type': 'text', 'value': 'GITHUB-EMBED GITHUB-EMBED'}
    ],
  };

const IMAGE_AST_ABSOLUTE = {
    'type': 'image',
    'url': 'https://www.google.com/images/dog.png',
    'children': [
      {'type': 'text', 'value': 'GITHUB-EMBED GITHUB-EMBED'}
    ],
};

const IMAGE_AST_RELATIVE = {
    'type': 'image',
    'url': '../images/dog.png',
    'children': [
      {'type': 'text', 'value': 'GITHUB-EMBED GITHUB-EMBED'}
    ],
};

const LINK_AST_ABSOLUTE = {
    'type': 'link',
    'url': 'https://www.google.com/images/dog.png',
    'children': [
      {'type': 'text', 'value': 'GITHUB-EMBED GITHUB-EMBED'}
    ],
};

const LINK_AST_RELATIVE = {
    'type': 'link',
    'url': '../images/dog.png',
    'children': [
      {'type': 'text', 'value': 'GITHUB-EMBED GITHUB-EMBED'}
    ],
};

const GRAPH_QL_PARENT_NODE = {
    internal: {
        type: 'node',
        mediaType: 'text',
    },
    parent: null,
};
const MARKDOWN_NODE = {
    internal: {
        type: 'node',
        mediaType: 'text',
    },
    parent: null,
};
module.exports = {
    PARAGRAPH_AST,
    IMAGE_AST_ABSOLUTE,
    IMAGE_AST_RELATIVE,
    LINK_AST_ABSOLUTE,
    LINK_AST_RELATIVE,
    GRAPH_QL_PARENT_NODE,
    MARKDOWN_NODE,
};
