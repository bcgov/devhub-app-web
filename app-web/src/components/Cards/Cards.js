import React from 'react';
import shortid from 'shortid';
import PropTypes from 'prop-types';
import styles from './Cards.module.css';
import Toggle from './Toggle';
import { CARD_TOGGLE_LIMIT, RESOURCE_TYPES, CARDS_PER_ROW } from '../../constants/ui';

/**
 * @param {Number} limit the maximum limit of cards to show in toggle component
 * @param {Number} numCards the number of cards to render
 * @param {Number} cardsPerRow
 * @returns {Number} the idea limit
 */
export const getIdealCardsLargeLimit = (limit, numCards, cardsPerRow) => {
  let newLimit = limit;
  while (numCards < newLimit && newLimit - cardsPerRow > 0) {
    newLimit -= cardsPerRow;
  }
  return newLimit;
};

const Cards = ({ topic, description, sourcePath, cards }) => {
  const cardComponents = cards.map(c => {
    return <p>yo </p>;
  });

  if (cardComponents.length > 0) {
    // find the ideal large screen limit based on number of cards, and cards rendering per row
    // ideally in a true grid system we would have a more concrete styling
    // to ensure cards are 3 per row. When this container is transferred to bootsraps grid system, this
    // will be so.
    const idealLimit = getIdealCardsLargeLimit(
      CARD_TOGGLE_LIMIT.LARGE,
      cardComponents.length,
      CARDS_PER_ROW,
    );

    return (
      <section className={styles.CardsContainer}>
        <div className={styles.TopicContainer}>
          <h1>{topic}</h1>
          <p>{description}</p>
        </div>
        <div className={styles.LargeView}>
          <Toggle cardComponents={cardComponents} cardLimits={idealLimit} />
        </div>
        <div className={styles.MobileView}>
          <Toggle cardComponents={cardComponents} cardLimits={CARD_TOGGLE_LIMIT.SMALL} />
        </div>
      </section>
    );
  } else {
    return null;
  }
};

Cards.defaultProps = {
  sourcePath: '',
  description: '',
};

Cards.propTypes = {
  cards: PropTypes.arrayOf(PropTypes.object).isRequired,
  sourcePath: PropTypes.string,
  topic: PropTypes.string.isRequired,
  description: PropTypes.string,
};

export default Cards;
