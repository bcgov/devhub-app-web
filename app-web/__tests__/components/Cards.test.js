import React from 'react';
import { shallow } from 'enzyme';
import Cards from '../../src/components/Cards/Cards';
import { COLLECTIONS } from '../../__fixtures__/siphon-fixtures';

describe('Card Component', () => {
  const CARDS = {
    DEFAULT: COLLECTIONS[0].nodes[0],
    REPOSITORY: COLLECTIONS[0].nodes[0],
    SELF_SERVICE: COLLECTIONS[0].nodes[0],
    COMPONENT: COLLECTIONS[0].nodes[0],
    DOCUMENT: COLLECTIONS[0].nodes[0],
  };

  it('matches snapshot', () => {
    const wrapper = shallow(<Cards cards={[CARDS.DEFAULT]} topic="topic" />);
    expect(wrapper).toMatchSnapshot();
  });
});
