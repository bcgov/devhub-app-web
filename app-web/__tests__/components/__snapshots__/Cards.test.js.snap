// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card Component matches snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Cards
    cards={
      Array [
        Object {
          "_metadata": Object {
            "position": "000.000.000",
          },
          "attributes": Object {
            "personas": Array [
              "Designer",
            ],
          },
          "childMarkdownRemark": Object {
            "frontmatter": Object {
              "pageOnly": false,
            },
          },
          "id": "1",
          "parent": Object {
            "id": "collection-1",
          },
          "resource": Object {
            "path": "/design-system/Z1bGIqu",
            "type": "Components",
          },
          "source": Object {
            "displayName": "Design System",
          },
          "unfurl": Object {
            "author": "ksingbeil",
            "data1": "",
            "data2": "",
            "description": "What the Design System is and how it works.",
            "image": "https://github.com/bcgov/design-system/blob/master/components/about/about.md?raw=true",
            "label1": "",
            "label2": "",
            "title": "About",
            "type": "markdown",
          },
        },
      ]
    }
    description=""
    sourcePath=""
    topic="topic"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="TopicContainer"
        >
          <h1>
            topic
          </h1>
          <p>
            
          </p>
        </div>,
        <div
          className="LargeView"
        >
          <Toggle
            cardComponents={
              Array [
                <p>
                  yo 
                </p>,
              ]
            }
            cardLimits={3}
          />
        </div>,
        <div
          className="MobileView"
        >
          <Toggle
            cardComponents={
              Array [
                <p>
                  yo 
                </p>,
              ]
            }
            cardLimits={1}
          />
        </div>,
      ],
      "className": "CardsContainer",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h1>
              topic
            </h1>,
            <p>
              
            </p>,
          ],
          "className": "TopicContainer",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "topic",
            },
            "ref": null,
            "rendered": "topic",
            "type": "h1",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "",
            },
            "ref": null,
            "rendered": "",
            "type": "p",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <Toggle
            cardComponents={
              Array [
                <p>
                  yo 
                </p>,
              ]
            }
            cardLimits={3}
          />,
          "className": "LargeView",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "cardComponents": Array [
              <p>
                yo 
              </p>,
            ],
            "cardLimits": 3,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <Toggle
            cardComponents={
              Array [
                <p>
                  yo 
                </p>,
              ]
            }
            cardLimits={1}
          />,
          "className": "MobileView",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "cardComponents": Array [
              <p>
                yo 
              </p>,
            ],
            "cardLimits": 1,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "div",
      },
    ],
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="TopicContainer"
          >
            <h1>
              topic
            </h1>
            <p>
              
            </p>
          </div>,
          <div
            className="LargeView"
          >
            <Toggle
              cardComponents={
                Array [
                  <p>
                    yo 
                  </p>,
                ]
              }
              cardLimits={3}
            />
          </div>,
          <div
            className="MobileView"
          >
            <Toggle
              cardComponents={
                Array [
                  <p>
                    yo 
                  </p>,
                ]
              }
              cardLimits={1}
            />
          </div>,
        ],
        "className": "CardsContainer",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h1>
                topic
              </h1>,
              <p>
                
              </p>,
            ],
            "className": "TopicContainer",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "topic",
              },
              "ref": null,
              "rendered": "topic",
              "type": "h1",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "",
              },
              "ref": null,
              "rendered": "",
              "type": "p",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Toggle
              cardComponents={
                Array [
                  <p>
                    yo 
                  </p>,
                ]
              }
              cardLimits={3}
            />,
            "className": "LargeView",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "cardComponents": Array [
                <p>
                  yo 
                </p>,
              ],
              "cardLimits": 3,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Toggle
              cardComponents={
                Array [
                  <p>
                    yo 
                  </p>,
                ]
              }
              cardLimits={1}
            />,
            "className": "MobileView",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "cardComponents": Array [
                <p>
                  yo 
                </p>,
              ],
              "cardLimits": 1,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
      ],
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
